code/modules/admin/topic.dm

```diff
@@ -2071,6 +2071,57 @@
 
 		usr << browse(dat.Join("<br>"), "window=related_[C];size=420x300")
 
+	else if(href_list["centcomlookup"])
+		if(!check_rights(R_ADMIN))
+			return
+
+		if(!CONFIG_GET(string/centcom_ban_db))
+			to_chat(usr, "<span class='warning'>Centcom Galactic Ban DB is disabled!</span>")
+			return
+
+		var/ckey = href_list["centcomlookup"]
+
+		var/list/query = json_decode(rustg_http_request_blocking(RUSTG_HTTP_METHOD_GET, "[CONFIG_GET(string/centcom_ban_db)][ckey]", null, null))
```

Comment:
Use the [http request helper datum](https://github.com/tgstation/tgstation/blob/master/code/datums/http.dm) and make it async please. Frankly, I never want to see this version of the function used. We don't need http calls locking up DD.